packages/ui-components/src/components/SearchInput/SearchInput.component.tsx

```diff
@@ -0,0 +1,226 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Juno contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState, useEffect, ChangeEvent, KeyboardEvent, MouseEvent } from "react"
+
+import { Icon } from "../Icon"
+import { Stack } from "../Stack"
+
+import "./searchinput.scss"
+
+export interface SearchInputProps {
+  /**
+   * Specifies the name attribute for the input element.
+   */
+  name?: string
+  /**
+   * Determines the visual styling variant of the SearchInput component.
+   * - "default": Standard search input styling.
+   * - "hero": A larger search input intended for standalone use on a dedicated search page, akin to the initial Google search page.
+   * - "rounded": A search input with rounded edges.
+   */
+  variant?: "rounded" | "hero" | "default"
+  /**
+   * Disables the search input when set to true.
+   */
+  disabled?: boolean
+  /**
+   * Custom placeholder text displayed in the search input.
+   */
+  placeholder?: string
+  /**
+   * Initial value for the search input. Note that this value will not be updated if changed by the user.
+   */
+  value?: string
```

Comment:
This line in the description "Note that this value will not be updated if changed by the user." is a bit confusing. If we set a value and the user types something else, it does change. Or do we mean that it's not changed permanently? Maybe we can remove?
<img width="516" alt="Screenshot 2024-10-17 at 10 17 27" src="https://github.com/user-attachments/assets/fde6d41c-456f-4e78-9a9a-f10b811e5f6c">

Reply:
yes you're right, this is outdated. What this was referring to was the fact that at some point in the lifetime of this component any updates to this value after the component was rendered wouldn't change the internal value of the input. But as you said, this is no longer the case because we ensure internal state update via the `useEffect` hook. So we could change the description as follows:

"Set the input value"